client/state/invites/selectors.js

```diff
@@ -69,6 +68,8 @@ export function getNumberOfInvitesFoundForSite( state, siteId ) {
 	return state.invites.counts[ siteId ] || null;
 }
 
+const findInvite = ( invites, inviteId ) =>
+	Array.isArray( invites ) ? invites.find( ( invite ) => invite.key === inviteId ) : undefined;
```

Comment:
`Array.prototype.find()` will return `undefined` if the element is not found, so we could simplify this to:

```suggestion
	invites?.find?.( ( invite ) => invite.key === inviteId );
```

Although I'm not sure: is there a case where `invites` will be non-nullish and will not be an array? If there's no such case, then we won't need the second optional chaining.

Furthermore, we can shorten by using destructuring:

```suggestion
	invites?.find?.( ( { key } ) => key === inviteId );
```

Reply:
I think you're right. If `siteInvites` exists and I'm reading the reducer correctly it always has `accepted` and `pending` arrays.